### PR TITLE
GEOMESA-2683 - Converter Metrics - AWS Cloudwatch Reporting

### PR DESCRIPTION
**Pull request by forrestfeaser**
Signed-off-by: Forrest Feaser <forrest.feaser@ccri.com>

### CODE REVIEW

File: geomesa-convert/geomesa-convert-metrics-cloudwatch/src/main/scala/org/locationtech/geomesa/convert/metrics/cloudwatch/CloudWatchReporterFactory.scala [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+  override def apply(
+      conf: Config,
+      registry: MetricRegistry,
+      rates: TimeUnit,
+      durations: TimeUnit): Option[ScheduledReporter] = {
+    if (!conf.hasPath("type") || !conf.getString("type").equalsIgnoreCase("cloudwatch")) { None } else {
+      val namespace = if(!conf.hasPath("namespace")) "geomesa" else conf.getString("namespace")
+      val reporter =
+        CloudWatchReporter.forRegistry(registry, AmazonCloudWatchAsyncClientBuilder.standard().withRegion(Regions.US_EAST_1).build(), namespace)
```
Comment (aheyne) [inline geomesa-convert/geomesa-convert-metrics-cloudwatch/src/main/scala/org/locationtech/geomesa/convert/metrics/cloudwatch/CloudWatchReporterFactory.scala]: We should pull the region into the conf.

File: docs/user/convert/parsing_and_validation.rst [hunk 2]
```diff
@@ -191,17 +191,26 @@ evaluation context, or can be exposed through reporters configured in the conver
...
     geomesa.converters.myconverter {
       options {
         reporters = {
+          cloudwatch = {
+            type           = "cloudwatch"
+            units          = "MILLISECONDS"
+            interval       = "60 seconds"
+            namespace      = "geomesa"
+            withZeroValues = true
+            withRawCounts  = false
```
Comment (elahrvivaz) [inline docs/user/convert/parsing_and_validation.rst]: could we change these two params to be dash-separated, to be consistent with the other params? e.g `with-raw-counts`, or even just `raw-counts`.

File: geomesa-convert/geomesa-convert-metrics-cloudwatch/src/main/scala/org/locationtech/geomesa/convert/metrics/cloudwatch/CloudWatchReporterFactory.scala [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+          .convertRatesTo(rates)
+          .convertDurationsTo(durations)
+      if(conf.hasPath("withZeroValues") && conf.getBoolean("withZeroValues")) reporter.withZeroValuesSubmission()
+      if(conf.hasPath("withRawCounts") && conf.getBoolean("withRawCounts")) reporter.withReportRawCountValue()
```
Comment (elahrvivaz) [inline geomesa-convert/geomesa-convert-metrics-cloudwatch/src/main/scala/org/locationtech/geomesa/convert/metrics/cloudwatch/CloudWatchReporterFactory.scala]: we should add `{}` around the if statement bodies, to be consistent with our style guide.

**Comment by elahrvivaz** [comment]
@forrestfeaser can you run the `build/calculate-cqs.sh` script and commit the changes? (the script should have been fixed with the last commit to master)

**Pull request closed without merging**